### DISCUSSION TITLE
Next.js features vs quality

### DISCUSSION THREAD

**Discussion by nocodehummel** [vercel/next.js#56696] [Polls]
Next.js is a great platform with an amazing community support. Unfortunately it happens too often that **things** break in new releases. Currently there are 2.4k unresolved issues reported.

In my case I cannot upgrade from 13.4.12 to 13.5.4 as the build breaks. 

I think it would make sense to slow down on development and stabilize the platform. In my opinion the unresolved issues are a risk for the popularity of the platform in the future.

**Comment by Elindo586** [comment]
Don't they keep it in trial period until it is stable? 

I don't usually "try" anything during its experimental period because I don't really have much time.

**Comment by nocodehummel** [comment]
I upgraded to release 14 which caused undocumented breaking changes! In my case headers that I relied upon are no longer part of the request.

**Comment by NicolasCharpentier** [comment]
TBH I feel the same. Loving Next features and studying if it could be the stack I need for a new app. 

There was a solution for every technical thing I needed and I could set it up (multi-tenant redirections, image optimization, dynamic site-map/robots), loved the app router and the way we configure components in it. 

But 
- when I hear friends experience, some youtubers, they always say stuff like it takes too many memory, it randomly crashes. 
- when i see open issues, damn, there's a lot.
- when i read the community complaints, i feel like you going too fast and making choices to just deliver.
- when i see solutions, saying to use instable_XX, i feel like you going too fast but maybe it's the way ? 

Maybe it's the way you shall move, delivering fast and incrementally, but i can't stop feeling it's at the expense of users. 

Anyway, i'm afraid of choosing Next for my new app, it will be big, have a lot of domain logic, pages. If I go this way and 
- i get crashes / memory problems.
- i get long builds cos lot of static pages.
- i get breaking changes frequently when updating Next.
I won't be happy. 

Having said that, I like the choices you made, especially in App Router and RSC, and don't see other frameworks that seem to me scalable enough for big apps (e.g., Remix might not have caching issues but the fetch-by-page is a no go for me in a typescript manner and components standalone-ity manner, for a big app)

**Comment by nocodehummel** [comment]
A year after this poll opened the number of open issues is >3K (2.4k previous). Bugs like e.g. link scroll and restoration are not prioritized. These type of bugs are annoying for the end-users of applications! The team is moving too fast for the next thing at the expense of quality.

**Comment by Ross249** [comment]
Yeah, I feel the same especially when you deploy your next.js project on other platforms except Vercel.
